services/api-dotnet/src/app/Api.NoSql/Controllers/UsersController.cs

```diff
@@ -64,8 +66,50 @@ public async Task<IActionResult> UpdateCurrentAsync([FromBody] UpdateCurrentUser
                 user.FirstName,
                 user.LastName,
                 user.Email,
-                user.IsEmailVerified
+                user.IsEmailVerified,
+                user.AvatarUrl
+            });
+        }
+
+        [HttpPost("upload-photo")]
+        public async Task<IActionResult> UploadCurrentPhoto([FromForm] UploadPhotoModel model)
+        {
+            await using var stream = model.File.OpenReadStream();
+
+            var avatarUrl = await _cloudStorageService.UploadPublicAsync(
+                $"avatars/{CurrentUserId}-{DateTime.UtcNow:s}-{model.File.FileName}",
+                stream
+            );
+
+            await _userService.UpdateAvatarUrlAsync(CurrentUserId, avatarUrl);
```

Comment:
1. To me it seems like this method should be responsible both for avatar upload and user update, especially considering ICloudStorageService is an infrastructure service and can be injected into IUserService.
2. As a result, more generic name (like UpdateAvatarAsync or SetAvatarAsync) should fit more.